packages/devtools-reps/src/reps/string.js

```diff
@@ -38,34 +39,50 @@ StringRep.propTypes = {
 function StringRep(props) {
   let {
     className,
-    cropLimit,
-    object: text,
-    member,
     style,
+    cropLimit,
+    object,
     useQuotes = true,
     escapeWhitespace = true,
+    member,
     openLink,
     omitLinkHref = true,
   } = props;
 
-  const classNames = ["objectBox", "objectBox-string"];
-  if (className) {
-    classNames.push(className);
-  }
-  let config = {className: classNames.join(" ")};
-  if (style) {
-    config.style = style;
-  }
+  let text = object;
 
-  if (useQuotes) {
-    text = escapeString(text, escapeWhitespace);
-  } else {
-    text = sanitizeString(text);
+  const shouldCrop = (!member || !member.open) && cropLimit && text.length > cropLimit;
+  const isLong = isLongString(object);
+
+  if (isLong) {
+    const {
+      fullText,
+      initial,
+      length,
+    } = object;
+
+    text = shouldCrop
+      ? initial.substring(0, cropLimit)
+      : fullText || initial;
+
+    if (text.length < length) {
+      text += "\u2026";
+    }
   }
 
-  const shouldCrop = (!member || !member.open) && cropLimit && text.length > cropLimit;
+  text = formatText({
+    useQuotes,
+    escapeWhitespace
+  }, text);
+
+  const config = getElementConfig({
+    className,
+    style,
+    actor: object.actor
+  });
+
   if (!containsURL(text)) {
-    if (shouldCrop) {
+    if (!isLong && shouldCrop) {
```

Comment:
Maybe we can add a comment here to say that longStrings crop is handled before

Reply:
@nchevobbe done!